libraries/ui/src/hooks/useCurrentTimeMs.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Returns the result of `Date.now()` on first render, updates every 30 seconds.
+ * This serves 2 purposes:
+ * - Avoid `purity` errors from eslint-plugin-react-hooks due to the time changing from
+ * render to render
+ * - Support live updating displays of time
+ */
+export const useCurrentTimeMs = () => {
+  const [currentTimeMs, setCurrentTimeMs] = useState(() => Date.now());
+
+  // Update current time every 30 seconds
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCurrentTimeMs(Date.now());
+    }, 30_000);
```

Comment:
Consider extracting this as a default parameter in the hook. Going forward it would be useful to exclusively use this hook for timers, some of which might need to be updated less/more frequently.